### DISCUSSION TITLE
Disable Descriptions from Doc-Strings

### DISCUSSION THREAD

**Discussion by RomeoAva** [fastapi/fastapi#11769] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
-
```

### Description

Hi, following this documentation: https://fastapi.tiangolo.com/tutorial/path-operation-configuration/?h=docstring#description-from-docstring

It is indicated that FastAPI can use the docstring as a description inside the Documentation (if a description is not specifically set on the endpoint itself).

I wanted to know if it was possible to disable that feature across a FastAPI app. Doc-strings can sometimes contain some what sensitive information (like backend information that I don't want the user to even know of). 

And if someone isn't aware that his doc-strings are being translated into public descriptions I would prefer this feature to not be on for my usecase.

I wasn't able to find anything about on how to disable this functionality. Thank you

### Operating System

macOS

### FastAPI Version

0.111.0

### Pydantic Version

2.7.4

### Python Version

Python 3.12.2

**Comment by YuriiMotov** [comment, marked as answer]
This logic is implemented here: https://github.com/tiangolo/fastapi/blob/e304414c93e68f0298ddd9e23af0abfd3b8cd4da/fastapi/routing.py#L492
And, as you can see there is no any way to disable it.

But you can create your own APIRoute class and redefine logic there:
```
from fastapi import FastAPI
from fastapi.routing import APIRoute

class MyRouteClass(APIRoute):
    def __init__(self, *args, **kwargs):
        kwargs['description'] = kwargs.get('description') or "-"
        super().__init__(*args, **kwargs)

app = FastAPI()
app.router.route_class = MyRouteClass

@app.get('/not-set')
def get_not_set():
    "Doc-strings can sometimes contain some what sensitive information"
    pass

@app.get('/explicitly', description="Explicitly set description")
def get_explicitly():
    "Doc-strings can sometimes contain some what sensitive information"
    pass
```
